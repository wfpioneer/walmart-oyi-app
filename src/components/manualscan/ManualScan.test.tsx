import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { fireEvent, render } from '@testing-library/react-native';
import ManualScanComponent from './ManualScan';

jest.mock('../../utils/scannerUtils.ts', () => ({
  manualScan: () => {}
}));

const mockDispatch = jest.fn();
jest.mock('react-redux', () => {
  const ActualReactRedux = jest.requireActual('react-redux');
  return {
    ...ActualReactRedux,
    useDispatch: () => mockDispatch
  };
});
jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'Icon');

describe('Test Location Manual Scan Component', () => {
  it('Renders Manual Scan component with a default keyboard type', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <ManualScanComponent
        keyboardType="default"
        placeholder="Test"
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  it('Renders Manual Scan component with a numeric keyboard type', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <ManualScanComponent
        keyboardType="numeric"
        placeholder="Test"
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('should set txtEntry value onchange and on clear btn click will clear the value', () => {
    const { getByTestId } = render(
      <ManualScanComponent
        keyboardType="default"
        placeholder="Test"
      />
    );
    const txtEntry = getByTestId('txtEntry');
    fireEvent.changeText(txtEntry, '1234');
    expect(txtEntry.props.value).toBe('1234');

    const clearbutton = getByTestId('clearbutton');

    fireEvent.press(clearbutton);
    expect(txtEntry.props.value).toBe('');
  });
});
