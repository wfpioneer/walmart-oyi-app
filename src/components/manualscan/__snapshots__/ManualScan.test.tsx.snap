// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Location Manual Scan Component Renders Manual Scan component with a default keyboard type 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffffff",
      "flexDirection": "row",
      "marginBottom": 8,
      "paddingRight": 16,
    }
  }
>
  <TextInput
    autoFocus={true}
    keyboardType="default"
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Test"
    selectionColor="#0071e9"
    style={
      Object {
        "color": "#000000",
        "flex": 1,
        "paddingLeft": 10,
      }
    }
    testID="txtEntry"
    value=""
  />
</View>
`;

exports[`Test Location Manual Scan Component Renders Manual Scan component with a numeric keyboard type 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffffff",
      "flexDirection": "row",
      "marginBottom": 8,
      "paddingRight": 16,
    }
  }
>
  <TextInput
    autoFocus={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Test"
    selectionColor="#0071e9"
    style={
      Object {
        "color": "#000000",
        "flex": 1,
        "paddingLeft": 10,
      }
    }
    testID="txtEntry"
    value=""
  />
</View>
`;
